setup.cfg: Replace deprecated license_file with license_files

mgorny opened the pull request:
Replace the deprecated `license_file` key with `license_files` to fix setuptools deprecation warning:

> The license_file parameter is deprecated, use license_files instead.

## Contribution checklist:

(also see [CONTRIBUTING.rst](../tree/master/CONTRIBUTING.rst) for details)

- [x] wrote descriptive pull request text
- [ ] added/updated test(s) (n/a)
- [ ] updated/extended the documentation (n/a)
- [ ] added relevant [issue keyword](https://help.github.com/articles/closing-issues-using-keywords/)
      in message body (n/a)
- [x] added news fragment in [changelog folder](../tree/master/docs/changelog)
  * fragment name: `<issue number>.<type>.rst` for example (588.bugfix.rst)
  * `<type>` is must be one of `bugfix`, `feature`, `deprecation`, `breaking`, `doc`, `misc`
  * if PR has no issue: consider creating one first or change it to the PR number after creating the PR
  * "sign" fragment with ```-- by :user:`<your username>`.```
  * please, use full sentences with correct case and punctuation, for example:
    ```rst
    Fixed an issue with non-ascii contents in doctest text files -- by :user:`superuser`.
    ```
  * also see [examples](../tree/master/docs/changelog)
- [x] added yourself to `CONTRIBUTORS` (preserving alphabetical order)

mgorny commented:
Do you do news fragments for trivial fixes?

gaborbernat reviewed (commented):
Anything user facing needs change log 👍

mgorny commented:
> Anything user facing needs change log +1

Is it user facing? Unless you're using an ancient version of setuptools, the result is exactly the same. `license_file` is a deprecated alias for `license_files`.

gaborbernat commented:
It might impact them so better be transparent about it. Stuff that doesn't require a change log it's like the ci files 👍

mgorny commented:
Ok, I've added the changelog entry, contributor entry, updated minimum `setuptools` version in `pyproject.toml` and bumped `setup-cfg-fmt` to hopefully stop it from readding `license_file`. Please let me know if you prefer that split over smaller commits.

mgorny commented:
Thanks!
